pages/guides/frontend-migrations/pf-4-to-pf5.mdx

```diff
@@ -0,0 +1,494 @@
+ @ ./src/components/SourceDetail/ApplicationsCard.js 20:0-50 180:71-87
+ @ ./src/pages/Detail.js 6:0-75 34:36-52
+ @ ./src/Routing.js 8:32-95
+ @ ./src/App.js 7:0-32 33:44-51
+ @ ./src/AppEntry.js 5:0-24 12:24-27
+ @ container entry ./RootApp[0]
+
+```
+
+The patched build checks each patternfly module and tries to resolve their source files. If it fails to find them, it will throw the above error. (The resolution is cached so it will not check one file more than once)
+
+There are two possible reasons for this:
+
+#### The PF migration was not done properly
+
+Some modules no longer exist, were renamed, or were moved to the `@patternfly/react-core/deprecated` directory. Make sure that you have properly migrated your source code and try again.
+
+#### The PF module does not follow naming conventions
+
+Some JS modules do not follow the naming convention so the patched build is unable to retrieve the file. There is a registry of overrides for this case. Please contact the `@crc-experience-services-team` on the `#forum-consoledot-ui` channel and the missing entry will be added.
+
+## Fixing tests
+
+This is probably the longest and most annoying part of the migration. The more tests you have the longer it will most likely take.
+
+We cannot really help with these but in general. Most tests that try to interact with the components will be using some combination of text or CSS selectors and almost all of these changed.
+
+Any snapshot tests will likely be broken as the DOM has changed for almost every component.
+
+### Babel is gone but it's still handy for tests
+
+Because the build now uses `ts-loader` only, the `babel` transpiler is no longer required. That said, if your application is not written fully in typescript and your testing environment is not set up for typescript, it is recommended to have a minimal babel setup for your unit tests.
+
+It is easier and more performant to use `babel-jest` (jest default transpiler) rather than `ts-jest` for your JavaScript files.
+
+This is an example of a minimal babel configuration to ensure your jest tests will run:
+
+```js
+// babel.config.js
+// Babel config is used only for jest tests
+module.exports = {
+  presets: [
+    '@babel/preset-env',
+    '@babel/preset-react',
+    '@babel/preset-typescript',
+  ],
+};
+```
+
+You only require these babel dependencies to ensure your test environment will successfully start up. 
+
+- `@babel/core`
+- `@babel/preset-env`
+- `@babel/preset-react`
+- `@babel/preset-typescript`
+
+### Updating babel config
+
+1. remove all babel related dependencies from your `package.json` and run `npm install` or `yarn`
+2. install only required babel dev dependencies with `npm i --save-dev @babel/core @babel/preset-env @babel/preset-react @babel/preset-typescript` or its yarn equivalent
+3. update your babel configuration file
+4. your test command should now run properly without any compilation errors (tests will still probably fail because of the PF migration changes)
```

Comment:
```suggestion
4. Your test command should now run properly without any compilation errors. (tests will still probably fail because of the PF migration changes)
```